identity-hub-did/src/main/java/org/eclipse/dataspaceconnector/identityhub/did/IdentityHubCredentialsVerifier.java

```diff
@@ -0,0 +1,107 @@
+import org.eclipse.dataspaceconnector.spi.result.AbstractResult;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.text.ParseException;
+import java.util.AbstractMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Obtains and verifies credentials associated with a DID.
+ * The DID document contains an IdentityHub service, the IdentityHubCredentialsVerifier gets credentials from the
+ * IdentityHub instance and verifies the credentials.
+ */
+public class IdentityHubCredentialsVerifier implements CredentialsVerifier {
+
+    private final IdentityHubClient identityHubClient;
+    private final Monitor monitor;
+    private final SignatureVerifier signatureVerifier;
+
+    /**
+     * Create a new credential verifier that uses an Identity Hub
+     *
+     * @param identityHubClient IdentityHubClient.
+     */
+    public IdentityHubCredentialsVerifier(IdentityHubClient identityHubClient, Monitor monitor, SignatureVerifier signatureVerifier) {
+        this.identityHubClient = identityHubClient;
+        this.monitor = monitor;
+        this.signatureVerifier = signatureVerifier;
+    }
+
+    /**
+     * Get credentials from the IdentityHub of participant, verifies the credentials, and returns the successfully verified credentials.
+     * To be successfully verified, credentials needs to be signed by the specified issuer in the JWT.
+     *
+     * @param didDocument of a participant. The Did Document should contain an IdentityHub service.
+     * @return VerifiableCredentials.
+     */
+    @Override
+    public Result<Map<String, Object>> getVerifiedCredentials(DidDocument didDocument) {
+        var hubBaseUrl = getIdentityHubBaseUrl(didDocument);
+        if (hubBaseUrl.failed()) return Result.failure(hubBaseUrl.getFailureMessages());
+
+        var jwts = identityHubClient.getVerifiableCredentials(hubBaseUrl.getContent());
+        if (jwts.failed()) {
+            return Result.failure(jwts.getFailureMessages());
+        }
+        var verifiedClaims = jwts.getContent()
+                .stream()
+                .filter(signatureVerifier::isSignedByIssuer);
+        var claims = verifiedClaims.map(this::extractCredential)
+                .filter(AbstractResult::succeeded)
+                .map(AbstractResult::getContent)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        return Result.success(claims);
+    }
+
+    private Result<Map.Entry<String, Object>> extractCredential(SignedJWT jwt) {
+        try {
+            var jwtClaims = jwt.getJWTClaimsSet().getClaims();
+            var payload = (HashMap<String, Object>) jwt.getPayload().toJSONObject();
+            var vc = (Map<String, Object>) payload.get("vc");
+            var credentialId = vc.get("id");
+            if (credentialId == null) {
+                return Result.failure("Credential id is missing");
+            }
+            payload.putAll(jwtClaims);
```

Comment:
Not sure to understand this step. Doesn't `jwtClaims` contain already everything that we need?